components/ntp_background_images/browser/view_counter_model.cc

```diff
@@ -8,17 +8,29 @@
 #include "base/check.h"
 #include "base/logging.h"
 #include "base/rand_util.h"
+#include "base/time/time.h"
 #include "brave/components/ntp_background_images/browser/features.h"
 #include "brave/components/ntp_background_images/common/pref_names.h"
 #include "components/prefs/pref_service.h"
 
 namespace ntp_background_images {
 
+namespace {
+
+constexpr base::TimeDelta kCountsResetTimeDelay = base::Days(1);
+
+}  // namespace
+
 ViewCounterModel::ViewCounterModel(PrefService* prefs) : prefs_(prefs) {
   CHECK(prefs);
 
-  count_to_branded_wallpaper_ =
-      prefs->GetInteger(prefs::kCountToBrandedWallpaper);
+  // When browser is restarted or component is updated, we reset to "initial"
+  // count.
```

Comment:
When component is updated, this `ViewCounterModel`'s ctor is not called.
Instead `ViewCounterModel::Reset()` is called.

Reply:
thanks, I misunderstood that